first pass at groups & access docs (#2946)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/groups.md

---
title: "Groups"
id: "groups"
description: "When you define groups in dbt projects, you turn implicit relationships into an explicit grouping."
keywords:
  - groups access mesh
---

## About Groups 

> A group is collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams, by restricting [access to private](access) models

Reviewer: ```suggestion
A group is a collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams by restricting [access to private](access) models
```

> A group is a collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams by restricting [access to private](access) models

Reviewer: ```suggestion
A group is a collection of resources within a dbt project. Groups are named, and every group has an `owner`. They enable intentional collaboration within and across teams by restricting [access to private](access) models.
```

### Declaring a group

Groups are defined in `.yml` files, nested under a `groups:` key.

<File name='models/marts/finance/finance.yml'>

```yaml
groups:
  - name: finance
    owner:
      # 'name' or 'email' is required; additional properties allowed
      email: finance@jaffleshop.com
      slack: finance-data
      github: finance-data-team
```

</File>

### Adding a model to a group

To add a model to a group, add the `group` property to a model entry in a `.yml` file.

<File name='models/schema.yml'>

```yml
models:
  - name: model_name
    group: finance
```

</File>

### Referencing a model in a group

> By default, all models within a group are `protected` in access - meaning they can be referenced by downstream resources in _any_ group in the project using the [`ref`](ref) function. If a grouped model's `access` property is set to `private`, only resources within its group will be able to make references to it. 

Reviewer: ```suggestion
By default, all models within a group are `protected` in access - meaning they can be referenced by downstream resources in _any_ group in the project using the [`ref`](ref) function. If a grouped model's `access` property is set to `private`, only resources within its group can reference it. 
```
